code/modules/maps/tg/reader.dm

```diff
@@ -132,27 +136,83 @@ var/global/use_preloader = FALSE
 				bounds[MAP_MAXY] = max(bounds[MAP_MAXY], min(ycrd, world.maxy))
 
 			var/maxx = xcrdStart
+
+			// Assemble the grid of keys
+			var/list/key_list = list()
+			for(var/line in gridLines)
+				var/list/line_keys = list()
+				xcrd = 1
+				for(var/tpos = 1 to length(line) - key_len + 1 step key_len)
+					if(xcrd > world.maxx)
+						if(cropMap)
+							break
+						else
+							world.maxx = xcrd
+
+					if(xcrd >= 1)
+						var/model_key = copytext(line, tpos, tpos + key_len)
+						line_keys[++line_keys.len] = model_key
+						#ifdef TESTING
+						else
+							++turfsSkipped
+						#endif
+						CHECK_TICK
+					maxx = max(maxx, ++xcrd)
```

Comment:
This line makes all loaded maps think they are 1 wider than they actually are and causes loading new z-levels from full width/height dmms to fail, because it believes they are 1 wider than maxx but they are not.

It should be xcrd++ not ++xcrd